cts/search/mcm_test.go

```diff
@@ -19,24 +20,49 @@ func TestMCM(t *testing.T) {
 	require.True(t, len(res.Clusters) > 1)
 	cluster := res.Clusters[0]
 
-	userID := cts.GenerateCanonicalPrefixName()
-	userID = strings.Replace(userID, ":", "-", -1)
-	userID = strings.Replace(userID, "_", "-", -1)
+	userIDPrefix := cts.GenerateCanonicalPrefixName()
+	userIDPrefix = strings.Replace(userIDPrefix, ":", "-", -1)
+	userIDPrefix = strings.Replace(userIDPrefix, "_", "-", -1)
+
+	userIDs := []string{
+		userIDPrefix + "-0",
+		userIDPrefix + "-1",
+		userIDPrefix + "-2",
+	}
+
+	// Assign a single userID
+	{
+		_, err := client.AssignUserID(userIDs[0], cluster.ClusterName)
+		require.NoError(t, err)
+	}
 
-	// Assign the userID and
+	// Assign multiple userIDs
 	{
-		_, err := client.AssignUserID(userID, cluster.ClusterName)
+		_, err := client.AssignUserIDs(
+			[]string{userIDs[1], userIDs[2]},
+			cluster.ClusterName,
+		)
 		require.NoError(t, err)
 	}
 
-	// Check that userID was properly assigned (using get/search/list)
+	// Check that userIDs were properly assigned (using get/search/list)
 	{
 		cts.Retry(func() bool {
-			_, err := client.GetUserID(userID)
+			_, err := client.GetUserID(userIDs[0])
+			return err == nil
+		})
+
+		cts.Retry(func() bool {
+			_, err := client.GetUserID(userIDs[1])
+			return err == nil
+		})
+
+		cts.Retry(func() bool {
+			_, err := client.GetUserID(userIDs[2])
 			return err == nil
 		})
 
-		_, err := client.SearchUserIDs(userID)
+		_, err := client.SearchUserIDs(userIDs[0])
```

Comment:
Here you need to search every single user id, not just for one.

Reply:
Indeed, I've misread the specs, thanks for catching it!